docs/component-elmish.md

```diff
@@ -4,6 +4,15 @@
 
 Elmish is a library for building single page applications in F# applications in following the [model-view-update](https://www.elm-tutorial.org/en/02-elm-arch/cover.html) architecture made famous by [Elm](http://elm-lang.org).
 
+```mermaid
```

Comment:
Please don't add the most confusing diagram of all Elmish, here is [my reasoning](https://github.com/Zaid-Ajaj/the-elmish-book/issues/2#issuecomment-507401397)

Also there is no such thing as "HTML sent to Browser" and "Page request" could literally mean anything

Reply:
@Zaid-Ajaj do you have an example diagram that you think would better illustrate the architecture? 
Didn't find one in your Elmish book.
"Picture is worth a thousand of words" - I'm all in for adding diagrams that could help understand the concepts

Reply:
@theimowski  I don't have a diagram in book exactly because I think explaining the Elmish runtime in words an examples works a lot better than using an overly simplified and potentially misleading diagram. I know people have been using *this* diagram a lot but every time the terms and visuals are really confusing when you compare it to what Elmish and React are *really* doing.

Reply:
I see. Still it should be possible to create one that wouldn't be misleading - but maybe it's better to have none rather than one that would look too complex. 🤷

Reply:
I thought it would be a relatively simple diagram. It's the same one we published a long time ago and we've seen in training courses that it is more useful than not having it.

@Zaid-Ajaj it would be great to understand what you mean when you say what they are "really" doing. Is this more about having a deeper understanding or something else? The whole point of an abstraction is to provide a simple model that is easy to reason about, particularly for someone just starting out.

Reply:
> we've seen in training courses that it is more useful than not having it.

@isaacabraham Personally speaking, it caused more confusion than it clarifying the Elmish model which is a diagram is supposed to accomplish. That said, if you many people seem to grasp Elmish more easily from it, then you can forget what I said.  

> it would be great to understand what you mean when you say what they are "really" doing. Is this more about having a deeper understanding or something else?

It is about understanding the abstraction itself. Elmish has clear life-cycle events where different things happen. These events don't translate well to this MVU diagram. That is my point and I referenced in the comment above why that is the case

Reply:
@Zaid-Ajaj I'm sorry that I didn't understand first - otherwise I wouldn't have asked. The question is, do you need to know those life cycle events just to get a high level understanding of what the whole point of MVU is? I would say no, but that's just my opinion. No-one is saying that this diagram is the bible on Elmish or React - just that someone at a glance can get a feel for it.

Reply:
> do you need to know those life cycle events just to get a high level understanding of what the whole point of MVU is?

@isaacabraham No, you don't need to know for a high-level understanding.